policy-service/src/component-tests/java/uk/gov/gchq/palisade/component/policy/model/AuditablePolicyRecordResponseTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.component.policy.model;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.json.JsonTest;
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.boot.test.json.JsonContent;
+import org.springframework.boot.test.json.ObjectContent;
+import org.springframework.test.context.ContextConfiguration;
+
+import uk.gov.gchq.palisade.service.policy.model.AuditErrorMessage;
+import uk.gov.gchq.palisade.service.policy.model.AuditablePolicyRecordResponse;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static uk.gov.gchq.palisade.service.policy.ApplicationTestData.AUDIT_ERROR_MESSAGE;
+import static uk.gov.gchq.palisade.service.policy.ApplicationTestData.RESPONSE;
+
+@JsonTest
+@ContextConfiguration(classes = AuditablePolicyRecordResponseTest.class)
+class AuditablePolicyRecordResponseTest {
+
+    @Autowired
+    private JacksonTester<AuditablePolicyRecordResponse> jsonTester;
+
+    /**
+     * Grouped assertion test for a {@link AuditablePolicyRecordResponse} which holds a PolicyResponse and no exception.
+     * This is the scenario where the object represents a standard message that is to be passed onto the next service.
+     * Create the object with the builder and then convert to the Json equivalent.
+     * Takes the JSON Object, deserialises and tests against the original Object
+     *
+     * @throws IOException throws if the {@link AuditablePolicyRecordResponse} object cannot be converted to a JsonContent.
+     *                     This equates to a failure to serialise or deserialise the string.
+     */
+    @Test
+    void testAuditablePolicyRecordResponseSerialisingAndDeserialising() throws IOException {
+        AuditablePolicyRecordResponse auditablePolicyRecordResponse = AuditablePolicyRecordResponse.Builder.create().withPolicyResponse(RESPONSE).withNoErrors();
```

Comment:
This is a simple example of what I was talking about in the slack.  I appreciate the recursive comparison tells all.  The downside is when I mistype the parameter "resourceId". as  "resourceID", it will be the test that checks each parameter in the JSON Object and the Java Object have both been this done correctly and there is no small typo buried in there that will bite me.  The recursive tells me that something is wrong, but I have to work at it to what went wrong.  So when I write the code & tests, this find grained test is useful.  Once the code is right, the recursive is more useful as then all we want to know is did some other change break this code.   I think there is another plus is that a new programmer should be able to go to the tests and be able to understand what does the class do.  Here I think I would gravitate to the "overkill" version as it spells it out in detail.